src/script/cpp_api/s_async.h

```diff
@@ -114,12 +116,39 @@ class AsyncEngine {
 	u32 queueAsyncJob(std::string &&func, PackedValue *params,
 			const std::string &mod_origin = "");
 
+	/**
+	 * Replace an async job if possible or queue the job otherwise
+	 * @param oldId The ID of the job to replace
+	 * @param func Serialized lua function
+	 * @param params Serialized parameters
+	 * @return jobid The job is replaced or queued
+	 */
+	u32 replaceAsyncJob(const u32 &oldId, std::string &&func, std::string &&params,
+			const std::string &mod_origin = "");
+
+	/**
+	 * Replace an async job if possible or queue the job otherwise
+	 * @param oldId The ID of the job to replace
+	 * @param func Serialized lua function
+	 * @param params Serialized parameters (takes ownership!)
+	 * @return ID of queued job
+	 */
+	u32 replaceAsyncJob(const u32 &oldId, std::string &&func, PackedValue *params,
+			const std::string &mod_origin = "");
+
 	/**
 	 * Engine step to process finished jobs
 	 * @param L The Lua stack
 	 */
 	void step(lua_State *L);
 
+	/**
+	 * Get the maximum number of threads that can be used by the async environment
+	 */
+	unsigned int getThreadingCapacity() const {
+		return autoscaleMaxWorkers;
```

Comment:
I don't think this situation is relevant if we decide to not document it for modders (https://github.com/minetest/minetest/pull/14602#discussion_r1624118683), since this would then only be used by `/unittests`.

Reply:
This can happen even in the unittest case.
Just change it to return `MYMAX(workerThreads.size(), autoscaleMaxWorkers);`

Reply:
> Just change it to return `MYMAX(workerThreads.size(), autoscaleMaxWorkers);`

Done.